library/src/main/scala/scalan/linalgebra/Matrices.scala

```diff
@@ -285,6 +364,31 @@ trait Matrices extends Vectors with Math { self: ScalanCommunityDsl =>
       CompoundMatrix(rows, length)
     }
   }
+
+  trait ConstMatrixCompanion extends ConcreteClass1[ConstMatrix] with AbstractMatrixCompanion {
+    override def fromColumns[T: Elem](cols: Coll[AbstractVector[T]]): Matrix[T] = {
+      val numColumns = cols.length
+      val numRows = cols(0).length
+      val item = cols(0)(0)
```

Comment:
Yes, I thought about it, but `DenseFlatMatrix.*(matrix: Matrix[T])` method uses `matrix.companion.fromRows`.
I also found, that user can multiply a 3x3 matrix and a vector with length 4 and it returns a result with no errors or warnings, so code has several unsafe places that user must care about.